newIDE/app/src/ProjectManager/ProjectManagerMainMenu.js

```diff
@@ -0,0 +1,241 @@
+import ChevronArrowLeft from '../UI/CustomSvgIcons/ChevronArrowLeft';
+import { Trans } from '@lingui/macro';
+import { LineStackLayout } from '../UI/Layout';
+import { useResponsiveWindowSize } from '../UI/Responsive/ResponsiveWindowMeasurer';
+import { isNativeMobileApp } from '../Utils/Platform';
+import ElementWithMenu from '../UI/Menu/ElementWithMenu';
+import Text from '../UI/Text';
+import ChevronArrowRight from '../UI/CustomSvgIcons/ChevronArrowRight';
+
+type Props = {|
+  project: ?gdProject,
+  closeDrawer: () => void,
+  mainMenuCallbacks: MainMenuCallbacks,
+  buildMainMenuProps: BuildMainMenuProps,
+
+  // Props controlled by the parent component
+  // so we can adapt the layout based on the selection.
+  selectedMainMenuItemIndices: Array<number>,
+  setSelectedMainMenuItemIndices: (Array<number>) => void,
+|};
+
+const ProjectManagerMainMenu = ({
+  project,
+  closeDrawer,
+  mainMenuCallbacks,
+  buildMainMenuProps,
+  selectedMainMenuItemIndices,
+  setSelectedMainMenuItemIndices,
+}: Props) => {
+  const mainMenuItems = React.useMemo(
+    () =>
+      adaptFromDeclarativeTemplate(
+        buildMainMenuDeclarativeTemplate(buildMainMenuProps),
+        mainMenuCallbacks
+      ),
+    [buildMainMenuProps, mainMenuCallbacks]
+  );
+
+  const { isMobile } = useResponsiveWindowSize();
+  const shouldUseNativeMenu = !isNativeMobileApp() && !isMobile;
+
+  const visibleMenuItems = React.useMemo(
+    () => {
+      let displayedMenuItems = mainMenuItems;
+      for (const index of selectedMainMenuItemIndices) {
+        const menuItem = displayedMenuItems[index];
+        if (!menuItem || !menuItem.submenu) {
+          break;
+        }
+
+        displayedMenuItems = menuItem.submenu;
+      }
+
+      return displayedMenuItems;
+    },
+    [mainMenuItems, selectedMainMenuItemIndices]
+  );
+
+  const isNavigatingInsideSubMenu = selectedMainMenuItemIndices.length > 0;
+  const onBack = React.useCallback(
+    () => {
+      if (selectedMainMenuItemIndices.length === 0) {
+        return;
+      }
+
+      setSelectedMainMenuItemIndices(
+        selectedMainMenuItemIndices.slice(
+          0,
+          selectedMainMenuItemIndices.length - 1
+        )
+      );
+    },
+    [selectedMainMenuItemIndices, setSelectedMainMenuItemIndices]
+  );
+
+  const enhanceMenuItems = React.useCallback(
+    (menuItems: Array<MenuItemTemplate>): Array<MenuItemTemplate> =>
+      menuItems.map((menuItem, index) => {
+        if (menuItem.submenu) {
+          return {
+            ...menuItem,
+            submenu: enhanceMenuItems(menuItem.submenu),
+          };
+        }
+
+        if (menuItem.click) {
+          const originalClick = menuItem.click;
+          // $FlowFixMe - TS is not able to make the difference between checkbox & classic item.
+          const newMenuItem: MenuItemTemplate = {
+            ...menuItem,
+            click: () => {
+              originalClick();
+              closeDrawer();
```

Comment:
I made the choice to close the drawer every time an item is clicked, otherwise it stays open and it feels really weird (after creating a project or opening a new window...)